### ISSUE TITLE
Reconsider strong naming the assembly for the NuGet packages

### ISSUE THREAD

**Issue by lscorcia** [simple-odata-client/Simple.OData.Client#806]
Hi, we are evaluating the use of this library in order to replace access to Sharepoint data currently done with CSOM, which is not supported on net fw > 4.8. In the Sharepoint world, it is customary to strong name assemblies in order to avoid issues with assembly versioning and GAC copies.

I have seen issue #155 where you describe the ideological reason for not applying a strong name. I understand those, but I would urge you to reconsider. The world is a different place from 2015 (even the links you reported are dead now!) and while I agree that signing code is an exercise in futility, maintaining a clear binding between a distribute NuGet package and its source project is quite important. The guidelines for NuGet packages now explicitly encourage applying a strong name (https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/strong-naming#create-strong-named-net-libraries) . Signing keys do not require a password, you can just upload the .snk file to Github and have it shared with the world, keeping 100% OSS reproducibility of the package.
Basically, strong naming has now lost all of its former attachment to digital signatures and is just a unique name for a binary.

Thanks for your time and have a nice day.

**Comment by object** [comment]
@lscorcia 

Thank you for the suggestion. It's a long time since I've investigated what would be the right approach to library signing. I will give it a new thought and come back to you.

**Comment by robertmclaws** [comment]
Any chance we could make this happen? I would like to use this for the Batch testing in https://github.com/odata/restier, but our whole codebase has to be strong name signed, so using this means we can't presently compile.

Thanks!

**Comment by object** [comment]
@robertmclaws @lscorcia Sorry for taking so long to respond. I believe I've finally come to conclusion that it's okay to start signing it. I guess we need to bump a major version then because formally it's a major change.

I am on vacation until the end of the month. If you feel like doing the work on signing, you can send me a PR. Otherwise I will have to look into it when I am back.

**Comment by lscorcia** [comment]
Thank you @object . I sent the pull request as proposed.
